rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/parsetreetoast/bif.kt

```diff
@@ -1,33 +1,41 @@
-        this.bif_scan() != null -> this.bif_scan().toAst(conf)
-        this.bif_trim() != null -> this.bif_trim().toAst(conf)
-        this.bif_trimr() != null -> this.bif_trimr().toAst(conf)
-        this.bif_triml() != null -> this.bif_triml().toAst(conf)
-        this.bif_subst() != null -> this.bif_subst().toAst(conf)
-        this.bif_len() != null -> this.bif_len().toAst(conf)
-        this.bif_dec() != null -> this.bif_dec().toAst(conf)
-        this.bif_char() != null -> this.bif_char().toAst(conf)
-        this.bif_timestamp() != null -> this.bif_timestamp().toAst(conf)
-        this.bif_diff() != null -> this.bif_diff().toAst(conf)
-        this.bif_editc() != null -> this.bif_editc().toAst(conf)
-        this.bif_found() != null -> this.bif_found().toAst(conf)
-        this.bif_eof() != null -> this.bif_eof().toAst(conf)
-        this.bif_abs() != null -> this.bif_abs().toAst(conf)
-        this.bif_int() != null -> this.bif_int().toAst(conf)
-        this.bif_editw() != null -> this.bif_editw().toAst(conf)
-        this.bif_rem() != null -> this.bif_rem().toAst(conf)
-        this.bif_replace() != null -> this.bif_replace().toAst(conf)
-        else -> TODO(this.text + " " + toPosition(conf.considerPosition))
+    val position = toPosition(conf.considerPosition)
+    return enrichPossibleExceptionWith("Error at line ${position?.line()} - ${this.text}") {
+        when {
+            this.bif_elem() != null -> NumberOfElementsExpr(
+                this.bif_elem().expression().toAst(conf),
+                position = position
+            )
+            this.bif_lookup() != null -> this.bif_lookup().toAst(conf)
+            this.bif_xlate() != null -> this.bif_xlate().toAst(conf)
+            this.bif_scan() != null -> this.bif_scan().toAst(conf)
+            this.bif_trim() != null -> this.bif_trim().toAst(conf)
+            this.bif_trimr() != null -> this.bif_trimr().toAst(conf)
+            this.bif_triml() != null -> this.bif_triml().toAst(conf)
+            this.bif_subst() != null -> this.bif_subst().toAst(conf)
+            this.bif_len() != null -> this.bif_len().toAst(conf)
+            this.bif_dec() != null -> this.bif_dec().toAst(conf)
+            this.bif_char() != null -> this.bif_char().toAst(conf)
+            this.bif_timestamp() != null -> this.bif_timestamp().toAst(conf)
+            this.bif_diff() != null -> this.bif_diff().toAst(conf)
+            this.bif_editc() != null -> this.bif_editc().toAst(conf)
+            this.bif_found() != null -> this.bif_found().toAst(conf)
+            this.bif_eof() != null -> this.bif_eof().toAst(conf)
+            this.bif_abs() != null -> this.bif_abs().toAst(conf)
+            this.bif_int() != null -> this.bif_int().toAst(conf)
+            this.bif_editw() != null -> this.bif_editw().toAst(conf)
+            this.bif_rem() != null -> this.bif_rem().toAst(conf)
+            this.bif_replace() != null -> this.bif_replace().toAst(conf)
+            else -> TODO()
```

Comment:
This message is less expressive than before

Reply:
Well, I think that the description of the not implemented bif is already given by the enrichPossibleExceptionWith method